src/main/java/com/glencoesoftware/omero/ms/core/OmeroWebJDBCSessionStore.java

```diff
@@ -0,0 +1,210 @@
+
+
+/**
+ * An OMERO.web session store.
+ * @author Chris Allan <callan@glencoesoftware.com>
+ *
+ */
+public class OmeroWebJDBCSessionStore implements OmeroWebSessionStore{
+
+    private static final String SELECT_SESSION_SQL =
+        "select session_data from django_session where session_key = ?";
+
+    private static final org.slf4j.Logger log =
+            LoggerFactory.getLogger(OmeroWebJDBCSessionStore.class);
+
+    private JDBCClient client;
+    private Connection sync_connection;
+    public OmeroWebJDBCSessionStore(String url,
+        String user,
+        String password,
+        Vertx vertx) {
+        client = JDBCClient.createShared(vertx, new JsonObject()
+            .put("url", url)
+            .put("driver_class", "org.postgresql.Driver")
+            .put("max_pool_size", 30)
+            .put("user", user)
+            .put("password", password));
+
+        String full_url = url + "?user=" + user + "&password=" + password; 
+        Properties props = new Properties();
+        props.setProperty("user", user);
+        props.setProperty("password", password);
+        try{
+            sync_connection = DriverManager.getConnection(url, props);
+        }
+        catch (SQLException e){
+            throw new RuntimeException(e.toString());
+        }
+    }
+
+    private IConnector getConnectorFromSessionData(String sessionData){
+        String decodedSessionData =
+            StringUtil.fromBytes(Base64.getDecoder().decode(sessionData));
+        PyString pystring = Py.newString(decodedSessionData);
+        PyList hash_and_data = pystring.split(":", 1);
+        PyString data_str = new PyString((String) hash_and_data.get(1));
+        PyDictionary djangoSession =
+            (PyDictionary) cPickle.loads(data_str);
+        log.debug("Session: {}", djangoSession);
+        IConnector connector = (IConnector) djangoSession.get("connector");
+        return connector;
+    }
+
+    /**
+     * Retrieve the OMERO.web session's current
+     * <code>omeroweb.connector.Connector</code> synchronously.
+     * @param sessionKey Session key to retrieve a connector for.
+     * @return Connector instance or <code>null</code> if session lookup fails.
+     */
+    public IConnector getConnector(String sessionKey){
+        PreparedStatement st = null;
+        try {
+            st = sync_connection.prepareStatement(SELECT_SESSION_SQL);
+            st.setString(1, sessionKey);
+            java.sql.ResultSet rs = st.executeQuery();
+            if (!rs.next()){
+                //Nothing returned from query
+                return null;
+            }
+            else{
+                String sessionData = rs.getString(1);
+                return getConnectorFromSessionData(sessionData);
+            }
+        }
+        catch (SQLException e){
+        }
+        finally {
+            try{
+                st.close();
+            }
+            catch (SQLException e){
+                log.error(e.toString());
```

Comment:
slf4j can do this nicely with the construct something like `log.error("SQL error...", e)`.